File: sgilt-core/src/main/java/net/franzka/sgilt/core/evenement/dto/EventDetailDto.java
package net.franzka.sgilt.core.evenement.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record EventDetailDto(
        UUID id,
        String title,
        LocalDate date,
        String eventType,
        String ambiance,
        String ville,
        String lieu,
        String nbInvites,
        String coverUrl,
        String sharedNote,
        String description,
        String momentCle,
        String countdown,
        LocalDateTime lastUpdateDate,
        ClientInfoDto clientInfo
) {}

File: sgilt-front/app/data/evenement/dto/EventDetailDto.ts
/**
 * DTO — contrat de réponse de l'API GET /events/:id
 */
export interface EventDetailDto {
  id: string
  title: string
  date?: string
  eventType?: string
  ambiance?: string
  ville?: string
  lieu?: string
  nbInvites?: string
  coverUrl?: string | null
  sharedNote: string
  description?: string
  momentCle?: string
  countdown: 'serein' | 'proche' | 'imminent' | 'past'
  lastUpdateDate?: string | null
  clientInfo: {
    firstName: string
    lastName: string
    phone?: string | null
    email: string
  }
}

/**
 * DTO — corps de la requête PATCH /events/:id
 */
export interface EventPatchRequestDto {
  lieu?: string
  sharedNote?: string
  eventType?: string
  ambiance?: string
  ville?: string
  nbInvites?: string
  description?: string
  momentCle?: string
}
